src/liftdrag_plugin/advanced_liftdrag_plugin.cpp

```diff
@@ -0,0 +1,641 @@
+  ignition::math::Vector3d velInLDPlane = vel - vel.Dot(bodyY)*bodyY;
+
+  // Define stability frame: X is in-plane velocity, Y is the same as body Y,
+  // and Z perpendicular to both
+  ignition::math::Vector3d stabilityX = velInLDPlane;
+  stabilityX.Normalize();
+  ignition::math::Vector3d stabilityY = bodyY;
+  ignition::math::Vector3d stabilityZ = stabilityX.Cross(stabilityY);
+
+  double span = std::sqrt(this->area*this->AR); // Wing span
+  double mac = this->area/span; //Mean aerodynamic chord
+
+  //Get non-dimensional body rates. Gazebo uses ENU, so some have to be flipped
+  ignition::math::Vector3d body_rates = this->link->RelativeAngularVel();
+  double speed = vel.Length();
+  double p = body_rates.X()*span/(2*speed); //Non-dimensionalized roll rate
+  double q = -1*body_rates.Y()*mac/(2*speed); //Non-dimensionalized pitch rate
+  double r = -1*body_rates.Z()*span/(2*speed); //Non-dimensionalized yaw rate
+
+  if (bodyX.Dot(vel) <= 0.0){
+    // Only calculate lift or drag if the wind relative velocity is in the same direction
+    return;
+  }
+
+  //Compute angle of attack, alpha, using the stability and body axes
+  double stabXHoriz = stabilityX.Dot(bodyX);
+  double stabXVert = stabilityX.Dot(bodyZ);
+  this->alpha = atan2(stabXVert,stabXHoriz);
+
+  double sinAlpha = sin(this->alpha);
+  double cosAlpha = cos(this->alpha);
+
+  //Compute sideslip angle, beta
+  double velSW = vel.Dot(bodyY);
+  double velFW = vel.Dot(bodyX);
+  this->beta = (atan2(velSW,velFW));
+
+  //Compute dynamic pressure
+  double speedInLDPlane = velInLDPlane.Length();
+  double dyn_pres = 0.5 * this->rho * speedInLDPlane * speedInLDPlane;
+
+  // Compute CL at ref_pt, check for stall
+  double CL;
+
+   // Use a sigmoid function to blend pre- and post-stall models
+  double sigma = (1+exp(-1*this->M*(this->alpha-this->alphaStall))+exp(this->M*(this->alpha+this->alphaStall)))/((1+exp(-1*this->M*(this->alpha-this->alphaStall)))*(1+exp(this->M*(this->alpha+this->alphaStall)))); //blending function
+
+/*
+The lift coefficient (as well as all the other coefficients) are defined with the
+coefficient build-up method and using a quasi-steady approach. The first thing that is
+calculated is the CL of the wing in steady conditions (normal CL-alpha curve). Secondly,
+the effect of the roll, pitch, and yaw is added through the AVL stability coefficients.
+Finally, the effect of the control surfaces is added.
+*/
+/*
+The lift coefficient of the wing is defined as a combination of a linear function for
+the pre-stall regime and a combination of exponents a trigonometric functions for the
+post-stall regime. Both functions are blended in with the sigmoid function.
+CL_prestall = this->CL0 + this->CLa*this->alpha
+CL_poststall = 2*(this->alpha/abs(this->alpha))*pow(sinAlpha,2.0)*cosAlpha
+*/
+
+  CL = (1-sigma)*(this->CL0 + this->CLa*this->alpha) + sigma*(2*(this->alpha/abs(this->alpha))*pow(sinAlpha,2.0)*cosAlpha);
```

Comment:
probably this is not a big deal (negligible w.r.t. everything else in the simulator).. but isnt `pow()` more expensive than just squaring via multiplication generally? or do we use something like `-ffast-math` for the compilation? @Jaeyoung-Lim 

random stack overflow benchmarks for reference:
https://stackoverflow.com/questions/2940367/what-is-more-efficient-using-pow-to-square-or-just-multiply-it-with-itself